src/main/java/core/basesyntax/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int transactionQuantity = transaction.getQuantity();
+        int currentQuantity = shopServiceImpl.getQuantity(fruit);
+        if (transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {
+            if (transactionQuantity > currentQuantity) {

```

Comment:
According to the checklist, you should check that the result balance is positive before saving it in the Storage. In this case, you are checking if the transaction quantity is greater than the current quantity, but this is not exactly the same. Consider calculating the new quantity first and then checking if it is negative.